SplunkRumWorkspace/SplunkRum/SplunkRum/ScreenFrames.swift

```diff
@@ -0,0 +1,106 @@
+//
+/*
+Copyright 2021 Splunk Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+import QuartzCore
+import UIKit
+
+class ScreenFrames: NSObject {
+
+    private var isRunning = false
+    private var displayLink: CADisplayLink?
+
+    private var slowFrameThreshold: CFTimeInterval = 1.0 / 59.0
+    private var frozenFrameThreshold: CFTimeInterval = 700.0 / 1000.0
+
+    private var currentIteration: Int = 0
+    private var startedTime: CFTimeInterval = CACurrentMediaTime()
+
+    private var slowCount: Int = 0
+    private var frozenCount: Int = 0
+    private var isFirstIteration: Bool = true
+    private var previousTimestamp: CFAbsoluteTime = CACurrentMediaTime()
+
+    override init() {
+            super.init()
+            isRunning = false
+    }
+
+    func startTracking() {
+        isRunning = true
+        stopTracking() /// make sure to stop a previous running display link
+        let displayLink = CADisplayLink(target: self, selector: #selector(displayLinkCallback))
+        displayLink.add(to: .main, forMode: .common)
+        self.displayLink = displayLink
+    }
+
+    func stopTracking() {
+        isRunning = false
+        displayLink?.invalidate()
+        displayLink = nil
+    }
+
+    @objc func displayLinkCallback(_ displayLink: CADisplayLink) {
+
+         let currentTime: CFTimeInterval = CACurrentMediaTime()
+         if isFirstIteration {
+            previousTimestamp = displayLink.targetTimestamp
+            isFirstIteration = false
+         }
+         let duration = displayLink.timestamp - previousTimestamp
+         previousTimestamp = displayLink.timestamp
```

Comment:
I think we have to account for the application going background here. If the displaylink callback is not being invoked in background, we can have a case where the render time is huge.

Reply:
Application state is used to do this when an application comes from the background to the foreground.

Reply:
Can you elaborate?

Reply:
We modified the code based on your suggestion to manage the app going into the background. You can refer the code here: https://github.com/signalfx/splunk-otel-ios/pull/58/files#diff-ce6f5d7a4bad04dd0be70bbfda633a95a580402a8776c0fca2c0e859977248aaR53-R56